lib/src/webcrypto/webcrypto.ecdh.dart

```diff
@@ -14,43 +14,223 @@
 
 part of 'webcrypto.dart';
 
+/// Elliptic Curve Diffie-Hellman (ECDH) is a key agreement protocol that allows
+/// two parties to establish a shared secret over an insecure channel.
+/// 
+/// An [EcdhPrivateKey] instance represents a private key that can be used to
+/// derive a shared secret with a public key.
+/// 
+/// Instances of [EcdhPrivateKey] can be imported from:
+/// * PKCS8 Key using [EcdhPrivateKey.importPkcs8Key], and,
+/// * JSON Web Key using [EcdhPrivateKey.importJsonWebKey].
+/// 
+/// A random key pair can also be generated using [EcdhPrivateKey.generateKey].
+/// 
+/// **Example**
+/// ```dart
+/// import 'package:webcrypto/webcrypto.dart';
+/// 
+/// Future<void> main() async {
+///   // Generate a new key pair using the P-256 curve.
+///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+///   
+///   // Derive 256 bits from the private key using the public key.
+///   final derivedBits = await keyPair.privateKey.deriveBits(256, keyPair.publicKey);
```

Comment:
This isn't what ECDH is used for right?

```dart
// Alice generates a key-pair
final kpA = await EcdhPrivateKey.generateKey(EllipticCurve.p256);

// Bob generates a key-pair
final kpB = await EcdhPrivateKey.generateKey(EllipticCurve.p256);

// If Alice and Bob exchanges public keys, they can generate the same shared secret
final sharedSecretA = await pkA.privateKey.deriveBits(256, pkB.publicKey);
final sharedSecretB = await pkB.privateKey.deriveBits(256, pkA.publicKey);
assert(base64.encode(sharedSecretA) == base64.encode(sharedSecretB));
```

Or am I wrong here? maybe we should do a test to make sure we understand how this is supposed to work.

Reply:
Makes much more sense. I was only providing an example of how the function is used instead of how ECDH should be used. I'll update the example with this